### PR TITLE
Add a bridge between WebView JavaScript and native Android code.

### PR DESCRIPTION
**Pull request by VarunBarad**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description

This commit adds a bridge between Android WebView and the native Android code so that we can call native Android methods from inside JavaScript when it is running inside WebView.

This will later allow for porting more features to native code while still using the web-view for everything else.

## [optional] What gif best describes this PR or how it makes you feel?

hybrid combo

### CODE REVIEW

File: app/src/main/java/to/dev/dev_android/view/main/view/MainActivity.kt [hunk 1]
```diff
@@ -43,6 +45,7 @@ class MainActivity : BaseActivity<ActivityMainBinding>(), CustomWebChromeClient.
     private fun setWebViewSettings() {
         binding.webView.settings.javaScriptEnabled = true
         binding.webView.settings.domStorageEnabled = true
+        binding.webView.addJavascriptInterface(this.webViewBridge, "androidWebViewBridge")
```
Comment (sierisimo) [inline app/src/main/java/to/dev/dev_android/view/main/view/MainActivity.kt]: We don't actually need the `this` to pass the parameter. I would say the less verbose, the readable.

**Comment by sduduzog** [comment]
Yhis will open up so much room for versatility

File: app/src/main/java/to/dev/dev_android/util/AndroidWebViewBridge.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import android.content.Context
+import android.util.Log
+import android.webkit.JavascriptInterface
+
+/**
+ * This class currently is empty because more methods would be added to it
+ * when new bridge functionalities are added.
+ */
+class AndroidWebViewBridge(private val context: Context) {
```
Comment (sierisimo) [inline app/src/main/java/to/dev/dev_android/util/AndroidWebViewBridge.kt]: I will suggest that until needed… we should be careful with storing `Context`around

Comment (VarunBarad) [reply app/src/main/java/to/dev/dev_android/util/AndroidWebViewBridge.kt]: Good point, I was initially thinking of implementing a `showToast` method as sample so I put it there and then forgot to take it out later.

**Comment by VarunBarad** [comment]
@maestromac It is saying that "Merging is blocked - The base branch restricts merging to authorized users."
Can you look into it? If not you then who else should I tag?

**Comment by VarunBarad** [comment]
It might have blocked the merging because I pushed a commit to this branch after your approval. That commit was only for updating my fork's feature branch with the changes that were pushed on `master` here.

**Comment by maestromac** [comment]
Sorry @VarunBarad , I should've mention that you don't need to worry about the out-of-date warning because your branch did work locally...but there was indeed an issue with master. I resolved it but I shouldn't have ignored it in first place 😅

**Pull request merged**